admin/src/components/ModalForm/Collection/Filters/index.jsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+
+import {
+  Grid,
+  Typography,
+  Flex,
+  Button,
+} from '@strapi/design-system';
+import { useIntl } from 'react-intl';
+import { useDispatch } from 'react-redux';
+
+import SelectConditional from '../../../SelectConditional';
+import { onChangeContentTypes } from '../../../../state/actions/Sitemap';
+
+// eslint-disable-next-line arrow-body-style
+const Filters = (props) => {
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+  const [conditionCount, setConditionCount] = React.useState(0);
+  const {
+    modifiedState,
+    uid,
+    langcode,
+    contentTypes,
+  } = props;
+
+  React.useEffect(() => {
+    // get the initial condition count
+    const count = Object.keys(modifiedState.getIn([uid, 'filters'], []).toJS()).length;
+    setConditionCount(count);
+  }, [uid, langcode]);
+
+  const handleRemoveCondition = () => {
+    dispatch(onChangeContentTypes(uid, null, ['filters', conditionCount, 'field'], ''));
+    dispatch(onChangeContentTypes(uid, null, ['filters', conditionCount, 'operator'], ''));
+    dispatch(onChangeContentTypes(uid, null, ['filters', conditionCount, 'value'], ''));
+    setConditionCount(conditionCount - 1);
+  };
```

Comment:
The `handleRemoveCondition` function always removes the last filter. It would be neat if we can specify the filter we want to remove. Giving the end user more control on the filters interface.